openai.el

```diff
@@ -115,10 +119,15 @@ return KEY."
 Arguments CONTENT-TYPE, KEY, and ORG-ID are common request headers."
   (setq key (openai--resolve-key key))
   (open--alist-omit-null `(("Content-Type"        . ,content-type)
-                           ("Authorization"       . ,(if (or (null key)
+                           ,(if (equal openai-key-type :bearer)
```

Comment:
I prefer to use `pcase` over here.

```elisp
(pcase openai-key-type
   (:bbb       ...)
   (:azure-api ...)
   (_ (user-error "Invalid key type: %s" openai-key-type)))
```

This way, it will be easier for users to debug and know what they should do. Or even better:

```elisp
(if (or (null key)
         (string-empty-p key))
     ""
   (pcase openai-key-type
     (:bbb       (concat "Bearer " key))
     (:azure-api key)
     (_          (user-error "Invalid key type: %s" openai-key-type))))
```

Reply:
TIL about pcase! Thank you for the feedback!